libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1116,26 +1116,25 @@ class MapboxNavigation @VisibleForTesting internal constructor(
      * By default uses [MapboxRerouteController].
      */
     fun setRerouteController(rerouteController: RerouteController) {
-        val currentController = this.rerouteController
-        this.rerouteController = LegacyRerouteControllerAdapter(rerouteController)
-
-        if (currentController?.state == RerouteState.FetchingRoute) {
-            currentController.interrupt()
-            reroute()
-        }
+        setRerouteController(
+            LegacyRerouteControllerAdapter(rerouteController) as NavigationRerouteController?
+        )
     }
 
     /**
-     * Set [RerouteOptionsAdapter]. It allows to modify [RouteOptions] before a reroute request
-     * is sent if the default reroute controller is used. Pass `null` to clear the adapter.
+     * Set [NavigationRerouteController] that's automatically invoked when user is off-route.
+     *
+     * By default [MapboxRerouteController] is used.
+     * Pass `null` to disable the controller.
```

Comment:
Maybe?
```suggestion
     * Pass `null` to disable automatic reroute. A user will stay in `OFF_ROUTE` state until a new route is set or the user gets back to the route. 
```

Reply:
@mapbox/navnative , will native navigator get back to `TRACKING` state if reroute is disabled and user get back to the previous route?

Reply:
reroute implementation is on the SDK, so disabling one doesn't affect NN

Reply:
@RingerJK , I mean what happens if you disable reroute, go offroad, and then go back on the route. Will you be in a tracking state after that?

Reply:
yes, for sure, a route is persisted between SDK and NN if not set a new one